report/renderer/performance-category-renderer.js

```diff
@@ -211,73 +212,104 @@ export class PerformanceCategoryRenderer extends CategoryRenderer {
-      const scoreA = a.auditRef.result.scoreDisplayMode === 'informative'
-        ? 100
-        : Number(a.auditRef.result.score);
-      const scoreB = b.auditRef.result.scoreDisplayMode === 'informative'
-        ? 100
-        : Number(b.auditRef.result.score);
-      if (scoreA !== scoreB) return scoreA - scoreB;
-
-      // Overall impact is the estimated improvement to the performance score
-      if (a.overallImpact !== b.overallImpact) {
-        return b.overallImpact * b.guidanceLevel - a.overallImpact * a.guidanceLevel;
+    /**
+     * @param {FilterType} acronym
+     */
+    function refreshFilteredAudits(acronym) {
+      for (const audit of allInsights) {
+        if (acronym === 'All') {
+          audit.auditEl.hidden = false;
+        } else {
+          const shouldHide = audit.auditRef.result.metricSavings?.[acronym] === undefined;
+          audit.auditEl.hidden = shouldHide;
+        }
       }
 
-      // Fall back to the linear impact if the normal impact is rounded down to 0
-      if (
-        a.overallImpact === 0 && b.overallImpact === 0 &&
-        a.overallLinearImpact !== b.overallLinearImpact
-      ) {
-        return b.overallLinearImpact * b.guidanceLevel - a.overallLinearImpact * a.guidanceLevel;
+      diagnosticAudits.sort((a, b) => {
+        // If the score display mode is "metricSavings", the `score` will be a coarse indicator of the overall impact.
+        // Therefore, it makes sense to sort audits by score first to ensure visual clarity with the score icons.
+        const scoreA = a.auditRef.result.scoreDisplayMode === 'informative'
+          ? 100
+          : Number(a.auditRef.result.score);
+        const scoreB = b.auditRef.result.scoreDisplayMode === 'informative'
+          ? 100
+          : Number(b.auditRef.result.score);
+        if (scoreA !== scoreB) return scoreA - scoreB;
+
+        // If there is a metric filter applied, we should sort by the impact to that specific metric.
+        if (acronym !== 'All') {
+          const aSavings = a.auditRef.result.metricSavings?.[acronym] ?? -1;
+          const bSavings = b.auditRef.result.metricSavings?.[acronym] ?? -1;
+          if (aSavings !== bSavings) return bSavings - aSavings;
+        }
+
+        // Overall impact is the estimated improvement to the performance score
+        if (a.overallImpact !== b.overallImpact) {
+          return b.overallImpact * b.guidanceLevel - a.overallImpact * a.guidanceLevel;
+        }
+
+        // Fall back to the linear impact if the normal impact is rounded down to 0
+        if (
+          a.overallImpact === 0 && b.overallImpact === 0 &&
+          a.overallLinearImpact !== b.overallLinearImpact
+        ) {
+          return b.overallLinearImpact * b.guidanceLevel - a.overallLinearImpact * a.guidanceLevel;
+        }
+
+        // Audits that have no estimated savings should be prioritized by the guidance level
+        return b.guidanceLevel - a.guidanceLevel;
+      }).forEach(item => {
```

Comment:
small nit: could this `.foreach` be in a separate line for readability? unless this is generally preferred then it's ok :)